File: java/src/FileParser.java
import java.io.*;
import java.util.ArrayList;

/**
 * @author Siôn Griffiths - sig2@aber.ac.uk
 *         Date: 09/03/14
 *         Time: 12:52
 */
public class FileParser {

  /**The File instance to hold a reference to a file */
  private File file;

  /**A Buffered reader to facilitate reading of the file*/
  private BufferedReader bRead;

  /**A String to hold each line of the file, initialised to empty*/
  private String line ="";

  private ArrayList<String> sentences;

  public FileParser(){
    sentences = new ArrayList<String>();
  }

  public ArrayList<String> readFile(String  fileName){
    try {
      file = new File(fileName);
      bRead = new BufferedReader(new FileReader(file));

      while((line = bRead.readLine()) != null){
        sentences.add(line);
      }

      bRead.close();
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    } catch (IOException e) {
      e.printStackTrace();
    }

    return sentences;
  }



}

File: cpp/FileParser.cpp
/**
 * @file   FileParser.cpp
 * @Author Sion Griffiths (sig2@aber.ac.uk)
 * @date   March 2014
 * @brief  FileParser returns lines from file one by one.
 *   
 */
#include "FileParser.h"
#include <iostream>
#include <fstream>




using namespace std;

/**
 * Constructs a FileParser
 * @param fileName name of input file
 */
FileParser::FileParser(string fileName) {
    setFile(fileName);

}

FileParser::FileParser(const FileParser& orig) {
}

FileParser::~FileParser() {

}

/**
 * Sets the current input file
 * @param fileName name of input file
 */
void FileParser::setFile(string fileName) {

    in_file.open(fileName.c_str());
}

/**
 * Reads and returns a single line from file
 * @return line from file as string
 */
string FileParser::readLine() {
    
    string line;
    if (!in_file.eof()) {
        in_file >> line;

        return line;
    } else {

        in_file.close();
        return NULL;
    }
}
